File: app/src/main/java/com/androidacademy/msk/exerciseproject/screen/newsdetails/NewsDetailsView.java
package com.androidacademy.msk.exerciseproject.screen.newsdetails;

import android.support.annotation.NonNull;

import com.androidacademy.msk.exerciseproject.data.database.entity.DbNewsItem;
import com.arellomobile.mvp.MvpView;
import com.arellomobile.mvp.viewstate.strategy.SingleStateStrategy;
import com.arellomobile.mvp.viewstate.strategy.StateStrategyType;

public interface NewsDetailsView extends MvpView {

    @StateStrategyType(SingleStateStrategy.class)
    void showNewsDetails(@NonNull DbNewsItem newsItem);
}

File: app/src/main/java/com/androidacademy/msk/exerciseproject/screen/newsdetails/NewsDetailsView.kt
package com.androidacademy.msk.exerciseproject.screen.newsdetails

import com.androidacademy.msk.exerciseproject.data.database.entity.DbNewsItem
import com.arellomobile.mvp.MvpView
import com.arellomobile.mvp.viewstate.strategy.SingleStateStrategy
import com.arellomobile.mvp.viewstate.strategy.StateStrategyType

interface NewsDetailsView : MvpView {

    @StateStrategyType(SingleStateStrategy::class)
    fun showNewsDetails(newsItem: DbNewsItem)
}
